x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/extension/CrossClusterAccessSecurityExtension.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.transport.extension;
+
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xpack.security.authc.CrossClusterAccessAuthenticationService;
+import org.elasticsearch.xpack.security.support.ReloadableSecurityComponent;
+import org.elasticsearch.xpack.security.transport.CrossClusterAccessTransportInterceptor;
+import org.elasticsearch.xpack.security.transport.CrossClusterApiKeySigner;
+import org.elasticsearch.xpack.security.transport.CrossClusterApiKeySigningConfigReloader;
+
+/**
+ * Remote cluster security extension point which is based on cross-cluster API keys.
+ */
+public class CrossClusterAccessSecurityExtension implements RemoteClusterSecurityExtension, ReloadableSecurityComponent {
+
+    private final CrossClusterAccessAuthenticationService authenticationService;
+    private final CrossClusterAccessTransportInterceptor transportInterceptor;
+
+    private final CrossClusterApiKeySigningConfigReloader crossClusterApiKeySignerReloader;
+    private final CrossClusterApiKeySigner crossClusterApiKeySigner;
+
+    private CrossClusterAccessSecurityExtension(Components components) {
+        this.crossClusterApiKeySignerReloader = new CrossClusterApiKeySigningConfigReloader(
+            components.environment(),
+            components.resourceWatcherService(),
+            components.clusterService().getClusterSettings()
+        );
+        this.crossClusterApiKeySigner = new CrossClusterApiKeySigner(components.environment());
+        crossClusterApiKeySignerReloader.setApiKeySigner(crossClusterApiKeySigner);
+
+        this.authenticationService = new CrossClusterAccessAuthenticationService(
+            components.clusterService(),
+            components.apiKeyService(),
+            components.authenticationService()
+        );
+        this.transportInterceptor = new CrossClusterAccessTransportInterceptor(
+            components.settings(),
+            components.threadPool(),
+            components.authenticationService(),
+            components.authorizationService(),
+            components.securityContext(),
+            this.authenticationService,
+            this.crossClusterApiKeySigner,
+            components.licenseState()
+        );
+    }
+
+    @Override
+    public CrossClusterAccessTransportInterceptor getTransportInterceptor() {
+        return transportInterceptor;
+    }
+
+    @Override
+    public CrossClusterAccessAuthenticationService getAuthenticationService() {
+        return authenticationService;
+    }
+
+    @Override
+    public void reload(Settings settings) {
+        this.crossClusterApiKeySignerReloader.reload(settings);
```

Comment:
I'm leaving cross cluster credentials reloading still in `Security` class. Will followup on that when I get better understanding of which settings we will still have to rely upon.

https://github.com/elastic/elasticsearch/blob/e929c93ac04627c8d4d0ac47c45efcaa715f7fc5/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java#L2351-L2365